### ISSUE TITLE
Adversarial and discriminator losses used is LSGAN?

### ISSUE THREAD

**Issue by francislata** [kan-bayashi/ParallelWaveGAN#255]
I noticed that the loss used is LSGAN. However, if I was to train that is close to the original MelGAN paper, it wouldn't match since they specified they found slightly better improvements with hinge loss.

What's the best way to adding hinge loss to actually match the MelGAN's training method?

**Comment by OnceJune** [comment]
I tried in a tf version by replacing every mse loss by hinge loss and it works, so I think you can do similar modification in pytorch version.

**Comment by francislata** [comment]
@OnceJune Can you point me to that TF version?

For the multiple discriminators, did you divide the sum of the adversarial losses by the number of discriminators (like [here](https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/parallel_wavegan/bin/train.py#L264))?

**Comment by OnceJune** [comment]
@francislata 

> Can you point me to that TF version?

I wrote one since our env is in a lower tf version

> For the multiple discriminators, did you divide the sum of the adversarial losses by the number of discriminators (like here)?

yes, replace mse by hinge is the only change I made

**Comment by hdmjdp** [comment]
> I tried in a tf version by replacing every mse loss by hinge loss and it works, so I think you can do similar modification in pytorch version.
@OnceJune 
Does this change improve something？

**Comment by OnceJune** [comment]
@hdmjdp no, hinge loss didn't perform better than radam

**Comment by hdmjdp** [comment]
> @hdmjdp no, hinge loss didn't perform better than radam

radam is better than AdamW?

**Comment by OnceJune** [comment]
@hdmjdp sorry my mistake, i tried hinge loss and mse loss on my dataset, mse loss fits my set better.
I used RAdam optimizer, which also fits better in my dataset than Adam and SGD.
